### ISSUE TITLE
In-place by/combine

### ISSUE THREAD

**Issue by bkamins** [JuliaData/DataFrames.jl#2127]
Sometimes it is useful to modify a data frame in place by groups without producing an output (something like `foreach` vs `map` for collections). We might consider adding versions of `by`/`combine` that just process the function passed over groups but do not produce an output.

Call this function tentatively `by!`

The thing is that using a syntax like e.g.:
```
by!(df, :key, :v => v -> v .-= mean(v))
```
is type stable thus much faster than:
```
for sdf in groupby(df, :key)
    sdf.v .-= mean(sdf.v)
end
```
which is the current way to perform such an operation.

**Comment by nalimilan** [comment]
Maybe this should be the role of `groupby` + `select!`? Though `by` may be the right name for a convenience function doing that.

**Comment by bkamins** [comment]
A good point, `groupby` + `transform!` also 😄.

**Comment by bkamins** [comment]
Fixed by #2214
